api/v1alpha6/openstackcluster_types.go

```diff
@@ -0,0 +1,239 @@
+
+	// ControlPlaneSecurityGroups contains all the information about the OpenStack
+	// Security Group that needs to be applied to control plane nodes.
+	// TODO: Maybe instead of two properties, we add a property to the group?
+	ControlPlaneSecurityGroup *SecurityGroup `json:"controlPlaneSecurityGroup,omitempty"`
+
+	// WorkerSecurityGroup contains all the information about the OpenStack Security
+	// Group that needs to be applied to worker nodes.
+	WorkerSecurityGroup *SecurityGroup `json:"workerSecurityGroup,omitempty"`
+
+	BastionSecurityGroup *SecurityGroup `json:"bastionSecurityGroup,omitempty"`
+
+	Bastion *Instance `json:"bastion,omitempty"`
+
+	// FailureReason will be set in the event that there is a terminal problem
+	// reconciling the OpenStackCluster and will contain a succinct value suitable
+	// for machine interpretation.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the OpenStackCluster's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of
+	// OpenStackClusters can be added as events to the OpenStackCluster object
+	// and/or logged in the controller's output.
+	// +optional
+	FailureReason *capierrors.ClusterStatusError `json:"failureReason,omitempty"`
+
+	// FailureMessage will be set in the event that there is a terminal problem
+	// reconciling the OpenStackCluster and will contain a more verbose string suitable
+	// for logging and human consumption.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the OpenStackCluster's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of
+	// OpenStackClusters can be added as events to the OpenStackCluster object
+	// and/or logged in the controller's output.
+	// +optional
+	FailureMessage *string `json:"failureMessage,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=openstackclusters,scope=Namespaced,categories=cluster-api,shortName=osc
+// +kubebuilder:storageversion
```

Comment:
As v1alpha6 should be the default for storing the data, `// +kubebuilder:storageversion` has to be removed from the v1alpha5 files.

Reply:
Ah, totally must missed that yesterday. Thanks for the hint :)